### PR TITLE
Added json logging for check.

### PR DESCRIPTION
**Pull request by perjahn**
What does this PR change? What problem does it solve?
-----------------------------------------------------

Adding json logging to the check command, in a similar fashion how the init command outputs json.

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------

No

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added tests for all code changes.
- [x] I have added documentation for relevant changes (in the manual).
- [ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [ ] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

**Comment by perjahn** [comment]
How do I update the manual at readthedocs.io?

File: cmd/restic/cmd_check.go [hunk 1]
```diff
@@ -282,15 +292,19 @@ func runCheck(ctx context.Context, opts CheckOptions, gopts GlobalOptions, args
...
 	errChan = make(chan error)
 	var wg sync.WaitGroup
 
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		bar := newProgressMax(!gopts.Quiet, 0, "snapshots")
-		defer bar.Done()
+		if !gopts.JSON {
+			bar := newProgressMax(!gopts.Quiet, 0, "snapshots")
```
Comment (MichaelEischer) [inline cmd/restic/cmd_check.go:305]: With this change, `chkr.Structure` uses the wrong instance of the `bar` variable.

File: cmd/restic/cmd_check.go [hunk 2]
```diff
@@ -399,7 +413,17 @@ func runCheck(ctx context.Context, opts CheckOptions, gopts GlobalOptions, args
...
 	}
 
-	Verbosef("no errors were found\n")
+	if !gopts.JSON {
+		Verbosef("no errors were found\n")
+	} else {
+		status := checkSuccess{
+			MessageType: "checked",
+			Message:     "no errors were found",
+			ID:          repo.Config().ID,
```
Comment (MichaelEischer) [inline cmd/restic/cmd_check.go:422]: The `ID` and `Repository` URL aren't really relevant here and thus should be removed.

**Review by MichaelEischer — CHANGES_REQUESTED** [review]
Sorry to be blunt. The PR completely ignores all error messages (see all `Printf` statements). As a consequence the JSON output would break once the `check` command actually detects an error, thus making the current changes pointless. This is worse than the current state of no JSON output at all.

Thus either convert everything (we can discuss how to handle stderr) or drop the PR.

**Comment by darkdragon-001** [comment]
Thanks for working on this! I think a single summary output in the end might be enough for many use cases (including mine of running restic from a script). Although one should reliably receive (at least) one json message (in both success and failure cases).

**Comment by MichaelEischer** [comment]
> I think a single summary output in the end might be enough for many use cases (including mine of running restic from a script).

Its far simpler to just inspect the exit code of the `check` command. If it is `0` then everything is fine, otherwise there's some problem. That's exactly the same information currently provided by the JSON output.

**Comment by MichaelEischer** [comment]
Closing as this PR needs a full rework anyways, feel free to open a new one if you're interested in addressing the review feedback.

**Pull request closed without merging**